File: api/app/Models/Complaint.php
<?php

namespace App\Models;

class Complaint extends Model
{
    protected $fillable = [
        'description',
        'complaint_type_id',
        'picture_id',
        'album_id',
        'about_user_id',
        'from_user_id',
        'status',
    ];

    public function type() {
        return $this->belongsTo(ComplaintType::class, 'complaint_type_id');
    }
    public function picture() {
        return $this->belongsTo(Picture::class);
    }
    public function album() {
        return $this->belongsTo(Album::class);
    }
    public function fromUser() {
        return $this->belongsTo(User::class, 'from_user_id');
    }
    public function aboutUser() {
        return $this->belongsTo(User::class, 'about_user_id');
    }
}

File: PicsyncAdmin/PicsyncAdmin/Models/Complaint.cs
using Newtonsoft.Json;
using PicsyncAdmin.Converters;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace PicsyncAdmin.Models
{
    public class Complaint
    {
        public int Id { get; set; }
        public string? Type { get; set; }
        public int? Status { get; set; }
        public string? Description { get; set; }
        public required User  AboutUser { get; set; }
        public User? FromUser { get; set; }
        public string? Sign { get; set; }
        public Picture? Picture { get; set; }
        public Album? Album { get; set; }
        // Свойство для получения описания статуса
        public string StatusDescription
        {
            get
            {
                return Status switch
                {
                    1 => "Рассмотрена",
                    0 => "Отклонена",
                    _ => "Не рассмотрена" // Для null (или других значений, если status не инициализирован)
                };
            }
        }
    }
}
